Add a Salvage Design Document (#569)
space-wizards/docs: src/en/space-station-14/departments/cargo/salvage.md

**Salvage** is a subdepartment of Supply focused on material acquisition through scrapping abandoned ruined structures in space and mining mineral-rich asteroids. They also reclaim valuables for Cargo and abandoned technology for Science. In the process they have to contend with a variety of enemies ranging from Space Fauna (such as the humble Space Carp) to still-surviving automated defence systems, and as such are equipped to handle PvE situations with upgrading weapons and armor.

## Intended Experience
Salvage's vibes and duties should embody the fantasies of things like *Hardspace Shipbreaker* and *Deep Rock Galactic*, as well as the platonic ideals of scrapyards, shipbreaking, and space mining.

Salvage's gameplay should embody the feeling of a roguelike, offering round-to-round variety and inround progression. This will be done as a parallel incentive to the primary goal of Salvage - Material Acquisition.

Salvage should, IC, invoke feelings of camraderie and dependability between Salvagers as they are their only tethers to safety when out in the deep dark of space.

> Salvagers should explore abandoned structures that provide entertaining combat challenges parallel to station-improving resources. These structures should invoke deeper connection to the setting and are good places to explore lore and the themes of SS14.

Reviewer: ```suggestion
Salvage's vibes and duties should embody the fantasies of things like *Hardspace Shipbreaker* and *Deep Rock Galactic*, as well as the platonic ideals of scrapyards, shipbreaking, and space mining.

Salvage's gameplay should embody the feeling of a roguelike, offering round-to-round variety and in-round progression. This will be done as a parallel incentive to the primary goal of Salvage - Material Acquisition.

Salvage should, IC, invoke feelings of camaraderie and dependability between Salvagers as they are their only tethers to safety when out in the deep dark of space.

Salvagers should explore abandoned structures that provide entertaining combat challenges parallel to station-improving resources. These structures should invoke a deeper connection to the setting and are good places to explore lore and the themes of SS14.
```

In short:

> - Reclamation, camraderie, combat, tension, lore.

Reviewer: ```suggestion
- Reclamation, camaraderie, combat, tension, lore.
```

### Narrow Gameplay Definition
At the start of the round, Salvagers should engage with a mechanic that places a random structure near the station that the Salvagers embark upon to retrieve materials and engage with low-level dangers. The structure should disappear after a predictable, consistent time period.

The rewards from this should be minimal but enough to ensure a solid foundation to build off. Salvagers should be able to use this method later in the round if there is a major shortage of materials. The danger on these structures should be low to match. Salvagers should be able to use this method to begin upgrading their abilities and provide base-line benefits to the station.

After engaging with the first method, Salvagers should be empowered to investigate static pre-made structures picked from a random list that float just past the ATS. These structures should be slightly more challenging than the first method, and thus be slightly more rewarding. While anyone should theoretically be able to enter these structures, their rewards should only be truly accessible to Salvagers and their tools.

> After engaging with the first and second method, Salvagers should have access to a large static structure that shifts from round-to-round. This structure should inherently take multiple excursions to fully explore, and the desired rewards should not be achievable without multiple excursions. Each excursion should greatly benefit from having multiple Salvagers engaging with it. This structure should be the most dangerous, and present the greatest rewards.

Reviewer: ```suggestion
After engaging with the first and second methods, Salvagers should have access to a large static structure that shifts from round to round. This structure should inherently take multiple excursions to fully explore, and the desired rewards should not be achievable without multiple excursions. Each excursion should greatly benefit from having multiple Salvagers engaging with it. This structure should be the most dangerous, and present the greatest rewards.
```

Theoretically, high-skill Salvagers should be able to skip steps, while new Salvagers should be able to intuit the intended pathway without being told.

Salvagers going further afield should not be unhealthy to the roundflow, as mechanics should be put in place to keep them connected to the station, activities on the station, and threats from station-originating antagonists even when they are on the third method.

> Salvagers should be able to choose to specialize for Mining or Scrapping, and another Salvager specializing in a different method should not lock the other method out. The choice to Mine or Scrap should be akin to a Scientist choosing Xenoarchaeology or Anomalies.

Reviewer: ```suggestion
Salvagers should be able to choose to specialize in Mining or Scrapping, and another Salvager specializing in a different method should not lock the other method out. The choice to Mine or Scrap should be akin to a Scientist choosing Xenoarchaeology or Anomalies.
```

- Salvagers should not spend their rounds hunting powerful items & ignoring their job in the process
  - Any progression should feed into and from resource generation
- There should be no confusion on if Salvage is or isn't staffed.
  - If a Salvager dies on their job, it should be a reasonable expectation that this information is discoverable.
  - Salvages presence should be felt by the departments they are acquiring materials and items for.
  - Supply should be able to rely on Salvage for material generation.
### Current Flaws
Due to a history of poor direction, low foresight, and subpar attempts at parity between ss14 and ss13, Salvage in its current state presents elements that are often unhealthy for the game.

These elements are not intrinsic to the fantasy of Salvage, and most are not caused by the mechanics of the department. Instead, many elements are a result of active poor decision making and weak core gameplay elements.

#### Weak Core Gameplay Elements
Salvagers are the primary department interfacing with the core gameplay mechanic of Space Travel. Unfortunately, Space Travel in its current form is not healthy.

> Anyone who touches space becomes practically invisible, causing it to be a powerful tool for antagonism for little invenstment, both in marooning people and outmaneuvering Security. This causes an issue for Salvage as it effectively disconnects them from the round and results in higher odds for Round Removal by engaging with the core gameplay loops when compared to other jobs in the game.

Reviewer: ```suggestion
Anyone who touches space becomes practically invisible, causing it to be a powerful tool for antagonism for little investment, both in marooning people and outmaneuvering Security. This causes an issue for Salvage as it effectively disconnects them from the round and results in higher odds for Round Removal by engaging with the core gameplay loops when compared to other jobs in the game.
```

> There is no reason to return to the station once you have left. Air tanks can be stored in backpacks and Air canisters can be dragged, resulting in the sole timer being practically ignorable even without considering the compressive nature of Liquid Internals.

Reviewer: ```suggestion
There is no reason to return to the station once you have left. Air tanks can be stored in backpacks, and Air canisters can be dragged, resulting in the sole timer being practically ignorable even without considering the compressive nature of Liquid Internals.
```

> Without a secondary, unavoidable timer that actively encourages anyone to spend time on station, Space will continue to be uneccesarily powerful. Without a means of tracking anyone in space (such as tracking elements on a mass scanner, or suit coordinates working offgrid), any job that can access space will always have an element of unhealthy gameplay.

Reviewer: ```suggestion
Without a secondary, unavoidable timer that actively encourages anyone to spend time on the station, Space will continue to be unnecessarily powerful. Without a means of tracking anyone in space (such as tracking elements on a mass scanner, or suit coordinates working off-grid), any job that can access space will always have an element of unhealthy gameplay.
```

There are a handful of core elements that many Salvage elements are built on that have a weak technical foundation, such as Space Generation and Dungeon Code. The elements that use these are not inherently flawed or counter to a healthy Salvage department, and can themselves be brought in line. The amount of elements that use these shaky foundations is to such a degree that spending time interrogating the symptoms may not be the best use of development energy.

For example:
- Clipping occurs with the ATS, Magnet, and Ruins
- Expeditions utilize Dungeon Code & Biome generation

#### Misdirected Design

> The Salvage Job Board in its current implementation actively counters all commonly accepted duties of salvage by expecting Salvagers to sell material producing items (ore, scrap), and then expects Salvagers to purchase their rewards, the rewards which cost more than the monetary output of the tasks undergone to unlock them.

Reviewer: ```suggestion
The Salvage Job Board, in its current implementation, actively counters all commonly accepted duties of salvage by expecting Salvagers to sell material-producing items (ore, scrap), and then expects Salvagers to purchase their rewards, the rewards which cost more than the monetary output of the tasks undergone to unlock them.
```

Theoretically, a Job Board is an effective way to promote the desired gameplay of salvage (Material Acquisition, PvE, etc.) while awarding effective incentives *and* allowing for round-to-round variety. This should be investigated and implemented.

Cargo should not be allowed to purchase Rare Materials that Salvage should be expected to obtain. This disincentivises Salvagers from acquiring Materials, as often the rest of Supply simply undermines the effort Salvage has already undergone, creating a negative feedback loop inround and across rounds.

Ruins, Wrecks, and other Salvage dungeons should not provide PvP-intended weaponry and armor, or antagonistically-inclined tools. While certain types of items are inevitable to be mapped thematically, they should not be mapped in a manner which encourages Salvagers to ignore material acquisition in favour of "Gamer Loot".

> Salvage should not be encouraged to create a shuttle by the core gameplay loop of Salvage, as shuttles cost a lot of materials, present an opportunity cost where they could be spending that time gathering materials or interacting with the station, and once the shuttle is complete it promotes isolationism from Salvage. To that end, functioning shuttle parts should not be a part of the loot pool.

Reviewer: ```suggestion
Salvage should not be encouraged to create a shuttle by the core gameplay loop of Salvage, as shuttles cost a lot of materials, present an opportunity cost where they could be spending that time gathering materials or interacting with the station, and once the shuttle is complete, it promotes isolationism from Salvage. To that end, functioning shuttle parts should not be a part of the loot pool.
```

While planetary exploration is Really Cool, the current implementation of Expeditions promotes isolationism, requires a feature that costs a lot of materials, and rarely results in useful items returned to station.
  - Expeditions can be retooled to meet the standards laid out by this document.

## Justifying Salvage's existence
Salvage's continues existence Is justified. It serves as the primary arm for PvE and engaging with the Spacefaring mechanics.

There is a real, palpable desire for the fantasies that the Salvage job touches upon. Also, engaging with the core mechanics of Salvage and the fantasies found therein is highly rewarding to players because you are directly, materially, bettering the station and the experience of everyone else on it.

Mechanically, Salvaging is a far mor interesting way of material acquisiton than purchasing. There is a clear path from work put in to materials that feels incredible sastisfying and rewarding. The round-to-round variance and number-go-up gameplay has many players coming back consistently.

When Salvage is done well, everyone benefits and everyone enjoys. It is worth the effort to improve Salvage and refine the gameplay.

> The flaws present in the current iteration of Salvage are solveable. The cultural perceptions and issues are remediable. The concept is, well, Salvageable.

Reviewer: ```suggestion
Salvage's continued existence is justified. It serves as the primary arm for PvE and engaging with the Spacefaring mechanics.

There is a real, palpable desire for the fantasies that the Salvage job touches upon. Also, engaging with the core mechanics of Salvage and the fantasies found therein is highly rewarding to players because you are directly, materially, bettering the station and the experience of everyone else on it.

Mechanically, Salvaging is a far more interesting way of material acquisition than purchasing. There is a clear path from work put in to materials that feels incredible sastisfying and rewarding. The round-to-round variance and number-go-up gameplay have many players coming back consistently.

When Salvage is done well, everyone benefits, and everyone enjoys. It is worth the effort to improve Salvage and refine the gameplay.

The flaws present in the current iteration of Salvage are solvable. The cultural perceptions and issues are remediable. The concept is, well, Salvageable.
```
